app/main/views/suppliers.py

```diff
@@ -86,6 +89,116 @@ def download_agreement_file(supplier_id, framework_slug, document_name):
     return redirect(url)
 
 
+@main.route('/suppliers/<supplier_id>/countersigned-agreements/<framework_slug>/'
+            '<document_name>', methods=['GET'])
```

Comment:
Is the `document_name` necessary?  Seems like it's hardcoded everywhere and since the route already contains `countersigned-agreements` and the framework slug, it might be okay to lose it.